clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java

```diff
@@ -628,6 +628,24 @@ public static MemoryRecords withRecords(byte magic, long initialOffset, Compress
         return builder.build();
     }
 
+    // for testing only, create a new memory-records that contains duplicated
```

Comment:
Seems like it would be easier to create a record set with two batches... Here's an example:

```java
        ByteBuffer buffer = ByteBuffer.allocate(2048);
        MemoryRecordsBuilder builder = MemoryRecords.builder(buffer, magic, compression, TimestampType.CREATE_TIME, 0L);
        builder.append(10L, "1".getBytes(), "a".getBytes());
        builder.close();

        builder = MemoryRecords.builder(buffer, magic, compression, TimestampType.CREATE_TIME, 1L);
        builder.append(11L, "2".getBytes(), "b".getBytes());
        builder.append(12L, "3".getBytes(), "c".getBytes());
        builder.close();
        buffer.flip();
```
In any case, perhaps we can move this to `LogValidatorTest` since we are not likely to need it elsewhere?

Reply:
`builder.close();` would reset the position so that the next batch would overwrite the first one. Let me change a bit to make it work.

Reply:
It only resets the position of a dup. We use the pattern above in several existing test cases